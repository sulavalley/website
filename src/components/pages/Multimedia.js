import React from 'react';
import { Embed } from 'semantic-ui-react';
import '../../css/pages/Multimedia.css';

const Multimedia = () => (
  <center>
    <div className="video-player">
      <Embed
        url="https://www.youtube.com/embed/videoseries?list=UUffHDdNhOCZFvt-f_D4kxFA"
        placeholder="https://trello-attachments.s3.amazonaws.com/5aff552929693747ec0b088d/5aff5c23f3cddefc5870f975/34b908e23ae35dd9a286309203978edd/logo_MRT-VS-HD.png"
        iframe={{
          allowFullScreen: true
        }}
      />
    </div>
  </center>
);

export default Multimedia;
