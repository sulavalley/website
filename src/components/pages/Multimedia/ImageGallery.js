import React from 'react';
import { Modal, Form, Input, Button, Image, Segment } from 'semantic-ui-react';
import base from '../../../base';

class ImageGallery extends React.Component {
  state = {
    images: [],
    imageModalIsOpen: false,
    newImage: {
      url: '',
      title: ''
    }
  };

  componentDidMount() {
    base.fetch('images', {
      context: this,
      asArray: true,
      then(data) {
        this.setState({ images: data });
      }
    });
  }

  handleChange = event => {
    const { newImage } = this.state;
    this.setState({
      newImage: { ...newImage, [event.target.name]: event.target.value }
    });
  };

  handleSubmit = () => {
    const { newImage } = this.state;
    base
      .push('images', {
        data: { ...newImage }
      })
      .then(() => this.setState({ imageModalIsOpen: false }))
      .catch(() => console.error('Something went terribly wrong'));
  };

  render() {
    const { imageModalIsOpen, images } = this.state;
    const { currentUser } = this.props;
    return (
      <div>
        {currentUser && (
          <Button onClick={() => this.setState({ imageModalIsOpen: true })}>
            Nueva publicación
          </Button>
        )}
        <Segment basic>
          <Image.Group size="medium">
            {images.map((image, i) => (
              <Image rounded key={i} src={image.url} />
            ))}
          </Image.Group>
        </Segment>
        <ImageModal
          open={imageModalIsOpen}
          onClose={() => this.setState({ imageModalIsOpen: false })}
          onSubmit={this.handleSubmit}
          onChange={this.handleChange}
        />
      </div>
    );
  }
}

const ImageModal = ({ open, onClose, onSubmit, onChange }) => (
  <Modal closeIcon open={open} onClose={onClose}>
    <Modal.Header>Nueva publicación</Modal.Header>
    <Modal.Content>
      <Form onSubmit={onSubmit}>
        <Form.Field>
          <label>Título</label>
          <Input
            onChange={onChange}
            name="title"
            placeholder="Ingrese un título"
          />
        </Form.Field>
        <Form.Field>
          <label>URL</label>
          <Input
            onChange={onChange}
            name="url"
            placeholder="Ingrese un enlace a la imagen"
          />
        </Form.Field>
        <Button type="submit">Publicar</Button>
      </Form>
    </Modal.Content>
  </Modal>
);

export default ImageGallery;
