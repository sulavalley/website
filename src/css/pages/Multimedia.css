.video-player {
  width: 78vw;
}
.placeholder {
  margin-left: 0%;
}
