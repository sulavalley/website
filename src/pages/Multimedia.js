import React from 'react';

const Multimedia = () => {
    return (
        <div>
            <h1>Estas en Multimedia</h1>
        </div>
    );
}

export default Multimedia;