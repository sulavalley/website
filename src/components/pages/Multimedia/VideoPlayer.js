import React from 'react';
import { Embed } from 'semantic-ui-react';
import '../../../css/pages/Multimedia.css';
import logo from '../../../assets/images/logo.png';

const VideoPlayer = props => (
  <div className="video-player">
    <Embed
      url={props.url}
      placeholder={logo}
      iframe={{
        allowFullScreen: true
      }}
    />
  </div>
);

export default VideoPlayer;
