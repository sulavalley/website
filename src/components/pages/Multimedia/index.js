import React from 'react';
import { Tab } from 'semantic-ui-react';
import VideoPlayer from './VideoPlayer';
import ImageGallery from './ImageGallery';

const Multimedia = ({ currentUser }) => (
  <Tab
    panes={[
      {
        menuItem: 'Imágenes',
        render: () => (
          <Tab.Pane>
            <ImageGallery currentUser={currentUser} />
          </Tab.Pane>
        )
      },
      {
        menuItem: 'Vídeos',
        render: () => (
          <Tab.Pane>
            <center>
              <VideoPlayer />
            </center>
          </Tab.Pane>
        )
      }
    ]}
  />
);

export default Multimedia;
