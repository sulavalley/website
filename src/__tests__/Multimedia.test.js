import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Multimedia from './../pages/Multimedia';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Multimedia />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  shallow(<Multimedia />);
});

test('renders correctly', () => {
  const tree = renderer.create(<Multimedia />).toJSON();
  expect(tree).toMatchSnapshot();
});
