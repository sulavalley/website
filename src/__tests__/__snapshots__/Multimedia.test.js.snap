// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<center>
  <div
    className="video-player"
  >
    <div
      className="ui embed"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="video play icon"
      />
      <img
        className="placeholder"
        src="https://trello-attachments.s3.amazonaws.com/5aff552929693747ec0b088d/5aff5c23f3cddefc5870f975/34b908e23ae35dd9a286309203978edd/logo_MRT-VS-HD.png"
      />
    </div>
  </div>
</center>
`;
